src/lobby.rs

```diff
@@ -58,13 +66,14 @@ impl Lobby {
             games: inner
                 .games
                 .iter()
-                .map(|(game_id, players)| {
+                .map(|(game_id, lobby)| {
                     (
                         *game_id,
-                        players
+                        lobby
+                            .participants
+                            .clone()
                             .into_iter()
-                            .map(|participant| &participant.player)
-                            .cloned()
+                            .map(|participant| participant.player.clone())
```

Comment:
this should have been fine as it was (but this is fine too)

Reply:
```
error[E0515]: cannot return reference to local data `participant.player`
  --> src/lobby.rs:75:48
   |
75 | ...                   .map(|participant| &participant.player)
   |                                          ^^^^^^^^^^^^^^^^^^^ returns a reference to data owned by the current function

error[E0507]: cannot move out of `lobby.participants` which is behind a shared reference
  --> src/lobby.rs:72:25
   |
72 | /                         lobby
73 | |                             .participants
   | |_________________________________________^ move occurs because `lobby.participants` has type `std::collections::HashSet<types::Participant>`, which does not implement the `Copy` trait

error: aborting due to 2 previous errors
```

was the error I get as it was

Reply:
yeah, I missed that it was still into_iter() on the first pass through